src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,812 @@
+    { "WanderAboutPointRadius", &INI::Parse_Real, nullptr, offsetof(LocomotorTemplate, m_wanderAboutPointRadius) },
+    { "RudderCorrectionDegree", &INI::Parse_Real, nullptr, offsetof(LocomotorTemplate, m_rudderCorrectionDegree) },
+    { "RudderCorrectionRate", &INI::Parse_Real, nullptr, offsetof(LocomotorTemplate, m_rudderCorrectionRate) },
+    { "ElevatorCorrectionDegree", &INI::Parse_Real, nullptr, offsetof(LocomotorTemplate, m_elevatorCorrectionDegree) },
+    { "ElevatorCorrectionRate", &INI::Parse_Real, nullptr, offsetof(LocomotorTemplate, m_elevatorCorrectionRate) },
+    { nullptr, nullptr, nullptr, 0 },
+};
+// clang-format on
+
+LocomotorStore::~LocomotorStore()
+{
+    for (auto it = m_locomotorTemplates.begin(); it != m_locomotorTemplates.end(); it++) {
+        it->second->Delete_Instance();
+    }
+
+    m_locomotorTemplates.clear();
+}
+
+LocomotorTemplate *LocomotorStore::Find_Locomotor_Template(NameKeyType namekey)
+{
+    if (namekey == NAMEKEY_INVALID) {
+        return nullptr;
+    }
+
+    auto t = m_locomotorTemplates.find(namekey);
+
+    if (t != m_locomotorTemplates.end()) {
+        return (*t).second;
+    } else {
+        return nullptr;
+    }
+}
+
+const LocomotorTemplate *LocomotorStore::Find_Locomotor_Template(NameKeyType namekey) const
+{
+    if (namekey == NAMEKEY_INVALID) {
+        return nullptr;
+    }
+
+    auto t = m_locomotorTemplates.find(namekey);
+
+    if (t != m_locomotorTemplates.end()) {
+        return (*t).second;
+    } else {
+        return nullptr;
+    }
+}
+
+void LocomotorStore::Reset()
+{
+    auto it = m_locomotorTemplates.begin();
+
+    for (;;) {
+        if (!(it != m_locomotorTemplates.end())) {
+            break;
+        }
+
+        if (it->second->Delete_Overrides() == nullptr) {
+            m_locomotorTemplates.erase(it);
+        } else {
+            ++it;
+        }
+    }
+}
+
+Locomotor *LocomotorStore::New_Locomotor(const LocomotorTemplate *tmpl)
+{
+    return new Locomotor(tmpl);
+}
+
+LocomotorTemplate *LocomotorStore::New_Override(LocomotorTemplate *tmpl)
+{
+    if (tmpl == nullptr) {
+        return nullptr;
+    }
+
+    LocomotorTemplate *t = new LocomotorTemplate(*tmpl);
+    tmpl->Set_Next(t);
+    t->Set_Is_Allocated();
+    return t;
+}
+
+void LocomotorStore::Parse_Locomotor_Template_Definition(INI *ini)
+{
+    if (g_theLocomotorStore == nullptr) {
+        throw CODE_06;
+    }
+
+    bool found = false;
+    const char *name = ini->Get_Next_Token();
+    NameKeyType key = g_theNameKeyGenerator->Name_To_Key(name);
+
+    LocomotorTemplate *t = g_theLocomotorStore->Find_Locomotor_Template(key);
+    if (t != nullptr) {
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            t = g_theLocomotorStore->New_Override(static_cast<LocomotorTemplate *>(t->Friend_Get_Final_Override()));
+        }
+
+        found = true;
```

Comment:
Any reason
```c++
g_theLocomotorStore->m_locomotorTemplates[key] = t
```
is postponed to the end of the function? Can it not simply be here?